### PR TITLE
Renamed all setup methods

### PR DESCRIPTION
**Pull request by VivekGandhi54**
Fixes issue #164 - renames setupConnection, setupEdges, etc. to allocateMemory()
Moves edgSign from AllEdges to AllNeuroEdges

### CODE REVIEW

File: Simulator/Connections/Neuro/ConnStatic.cpp [hunk 1]
```diff
@@ -41,16 +41,16 @@ ConnStatic::~ConnStatic() {
 ///  @param  layout    Layout information of the neural network.
 ///  @param  vertices   The Vertex list to search from.
 ///  @param  edges  The Synapse list to search from.
-void ConnStatic::setupConnections(Layout *layout, IAllVertices *vertices, AllEdges *edges) {
+void ConnStatic::allocateMemory(Layout *layout, IAllVertices *vertices, AllEdges *edges) {
```
Comment (stiber) [inline Simulator/Connections/Neuro/ConnStatic.cpp:44]: Does this method do more than allocate memory?

File: Simulator/Core/GPUModel.cu [hunk 2]
```diff
@@ -82,13 +82,13 @@ void GPUModel::deleteDeviceStruct(void** allVerticesDevice, void** allEdgesDevic
 }
 
 /// Sets up the Simulation.
-void GPUModel::setupSim()
+void GPUModel::allocateMemory()
```
Comment (stiber) [inline Simulator/Core/GPUModel.cu:85]: Seems like this also does more than allocate memory.

File: Simulator/Core/Model.cpp [hunk 3]
```diff
@@ -68,17 +68,17 @@ void Model::createAllVertices() {
 }
 
 /// Sets up the Simulation.
-void Model::setupSim() {
+void Model::allocateMemory() {
```
Comment (stiber) [inline Simulator/Core/Model.cpp:71]: Seems that this does more than call memory allocation methods. At least, at the end, it creates the index map. That might be just a memory allocation method, however.

Comment (VivekGandhi54) [reply Simulator/Core/Model.cpp:71]: Yeah, you're right. There's some inconsistent usage of the setup methods, probably why they weren't called allocateMemory to begin with.
Should I undo all the renaming, or just keep the changes for setupEdges and setupVertices?

File: Simulator/Layouts/Layout.cpp [hunk 4]
```diff
@@ -66,7 +66,7 @@ shared_ptr<IAllVertices> Layout::getVertices() const {
 
 /// Setup the internal structure of the class.
 /// Allocate memories to store all layout state, no sequential dependency in this method
-void Layout::setupLayout() {
+void Layout::allocateMemory() {
```
Comment (stiber) [inline Simulator/Layouts/Layout.cpp:69]: I think this is an example where it's not pure memory allocation...

**Pull request closed without merging**